src/desugar/class.rs

```diff
@@ -0,0 +1,140 @@
+use crate::core::construct::Core;
+use crate::desugar::common::desugar_vec;
+use crate::desugar::context::Context;
+use crate::desugar::context::State;
+use crate::desugar::node::desugar_node;
+use crate::parser::ast::ASTNode;
+use crate::parser::ast::ASTNodePos;
+use std::ops::Deref;
+
+/// Desugar a class.
+///
+/// If a class has inline arguments (arguments next to class), then we create a
+/// constructor and assume that there is no constructor in the body of a class.
+/// This property should be ensured by the type checker.
+///
+/// We add arguments and calls to super for parents.
+pub fn desugar_class(node: &ASTNode, ctx: &Context, state: &State) -> Core {
```

Comment:
This is the important part of this PR.
The application logic for desugaring both inline constructors and calls to parent is somewhat convoluted at the moment. I'm lookinto ways to simplify this.
I could probably split this into two major passes, one to check what has to be built, and a second one that actually builds the relevant nodes.
As it is now, these two stages are a bit intertwined (though this is also somewhat necessary as we want to desugar default and literal arguments as well).